apps/slackbot/task_agent.py

```diff
@@ -87,6 +89,169 @@ def from_llm_and_tools(
             max_iterations=max_iterations,
         )
 
+    def chain_run(self, task, user_input):
+        try:
+            assistant_reply = self.action_planner.select_action(
+                self.previous_message, self.memory, task=task, user_input=user_input
+            )
+        except openai.error.APIError as e:
```

Comment:
It's great to see you're catching these exceptions / handling them gracefully. 

I think you should also log them, e.g. `logger.info`, because some of them like invalid request and exceeding rate limit represent things that should not happen in our app. I.e. we should modify our code in the future to prevent those exceptions from occurring. Right now we're just silently swallowing them.

Reply:
Agreed. Does it matter whether I use `logger.error` or `logger.info`?